app/components/traffic-measure/index.js

```diff
@@ -275,7 +275,8 @@ export default class TrafficMeasureIndexComponent extends Component {
   @task
   *saveRoadsigns(trafficMeasureConcept) {
     // delete existing ones
-    for (let i = 0; i < trafficMeasureConcept.relations.length; i++) {
+    let length = trafficMeasureConcept.relations.length;
+    for (let i = 0; i < length; i++) {
       const relation = trafficMeasureConcept.relations.objectAt(0);
```

Comment:
No because when destroying the record, `trafficMeasureConcept.relations` gets affected and is every time smaller. For example you'd have :
- Loop 1 : ['a', 'b', 'c'] -> destroy -> ['b', 'c']
- Loop 2 : ['b', 'c'] -> destroy -> ['c']
- Loop 3 : ['c'] -> destroy -> []

Reply:
ah fair enough, i guess when its a ember data array thats true, when its a javascript array i think it just sets the model objects to `.isDeleted=true`

> No because when destroying the record, `trafficMeasureConcept.relations` gets affected and is every time smaller. For example you'd have :
> 
> * Loop 1 : ['a', 'b', 'c'] -> destroy -> ['b', 'c']
> * Loop 2 : ['b', 'c'] -> destroy -> ['c']
> * Loop 3 : ['c'] -> destroy -> []

ah fair enough, i guess when its a ember data array thats true, when its a javascript array i think it just sets the model objects to `.isDeleted=true`